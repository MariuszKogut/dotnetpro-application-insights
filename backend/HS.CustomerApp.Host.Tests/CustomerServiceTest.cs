using FluentAssertions;
using HS.CustomerApp.Host.Logic;
using HS.CustomerApp.Host.Models;
using Snapshooter.Xunit;
using Xunit;

namespace HS.CustomerApp.Host.Tests
{
    public class CustomerServiceTest
    {
        [Fact]
        public void ShouldReturnAllCustomers()
        {
            // Arrange
            var sut = new CustomerService();

            // Act
            var result = sut.ReadAll();

            // Assert
            Snapshot.Match(result);
        }

        [Theory]
        [InlineData(1, "Microsoft")]
        [InlineData(2, "Oracle")]
        [InlineData(3, "IBM")]
        public void ShouldReturnSingleCustomer(long id, string expectedName)
        {
            // Arrange
            var sut = new CustomerService();

            // Act
            var result = sut.Read(id);

            // Assert
            result.Id.Should().Be(id);
            result.Name.Should().Be(expectedName);
        }

        [Fact]
        public void ShouldDeleteCustomer()
        {
            // Arrange
            var sut = new CustomerService();

            // Act
            sut.Delete(1);

            // Assert
            var item = sut.Read(1);
            item.Should().BeNull();
        }

        [Fact]
        public void ShouldAddCustomer()
        {
            // Arrange
            var sut = new CustomerService();
            var customer = new CustomerModel
            {
                Id = 4711,
                Name = "Facebook",
                Location = "USA"
            };

            // Act
            sut.Create(customer);

            // Assert
            var item = sut.Read(4711);
            item.Should().Be(customer);
        }

        [Fact]
        public void ShouldUpdateCustomer()
        {
            // Arrange
            var sut = new CustomerService();
            var customer = new CustomerModel
            {
                Id = 4711,
                Name = "Facebook",
                Location = "USA"
            };
            sut.Create(customer);

            // Act
            var updatedCustomer = new CustomerModel
            {
                Id = 4711,
                Name = "Facebook Inc",
                Location = "Menlo Park, California, USA"
            };
            sut.Update(updatedCustomer);

            // Assert
            var item = sut.Read(4711);
            item.Should().Be(updatedCustomer);
        }
    }
}