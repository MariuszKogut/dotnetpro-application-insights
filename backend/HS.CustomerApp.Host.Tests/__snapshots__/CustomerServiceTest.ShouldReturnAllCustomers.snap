[
  {
    "Id": 1,
    "Name": "Microsoft",
    "Location": "USA"
  },
  {
    "Id": 2,
    "Name": "Oracle",
    "Location": "USA"
  },
  {
    "Id": 3,
    "Name": "IBM",
    "Location": "USA"
  },
  {
    "Id": 4,
    "Name": "SAP",
    "Location": "Germany"
  },
  {
    "Id": 5,
    "Name": "Synamtec",
    "Location": "USA"
  },
  {
    "Id": 6,
    "Name": "EMC",
    "Location": "USA"
  },
  {
    "Id": 7,
    "Name": "VMWare",
    "Location": "USA"
  },
  {
    "Id": 8,
    "Name": "HP",
    "Location": "USA"
  },
  {
    "Id": 9,
    "Name": "Salesforce.com",
    "Location": "USA"
  },
  {
    "Id": 10,
    "Name": "Intuit",
    "Location": "USA"
  }
]
